src/sidebar/generate-ics-pane.jsx

```diff
@@ -24,24 +24,40 @@ export default class GenerateICSPane extends React.Component {
     }
 
     getFeedUrl() {
+        // In order to copy the URL as a direct response to the button click,
+        // a temporary ID is generated, and preemptively copied, but the success
+        // message is not displayed until the request returns. If the response
+        // has a different URL for some reason, it is copied again, requiring
+        // the user to press "control-C".
+
         var labels = this.props.selectedLabels;
-        var url = window.abe_url + '/subscriptions/'
+
+        var temp_id = Math.random().toString(16).substring(2, 15) + Math.random().toString(16).substring(2, 15);
```

Comment:
can also say `Math.random().toString(16).substring(2) + Math.random().toString(16).substring(2)`.

Reply:
I ended up changing it to .substring(2,12) because I'm not sure the length of the generated decimal is consistent across Javascript implementations, and I want the generated ID's to be robust against that.